packages/core/src/getConfig.ts

```diff
@@ -0,0 +1,39 @@
+import { getBrowserJestConfig } from './browser/config/getBrowserJestConfig'
+import { getNodeJestConfig } from './node/config/getNodeJestConfig'
+
+// TODO: Add more types like eslint, babel, etc?
+// TODO: Move somewhere else
+enum ConfigType {
+  Jest = 'jest',
+}
+
+// TODO: Add more build targets
+// TODO: Move somewhere else
+enum BuildTarget {
+  Browser = 'browser',
+  Node = 'node',
+}
+
+interface GetJestConfigParams {
+  type: ConfigType.Jest
+  target: BuildTarget
+}
+
+const jestConfigMap = {
+  [BuildTarget.Browser]: getBrowserJestConfig,
+  [BuildTarget.Node]: getNodeJestConfig,
+}
+
+function getJestConfig({ target }: GetJestConfigParams) {
+  return jestConfigMap[target]()
+}
+
+const configMap = {
+  jest: getJestConfig,
+}
+
+type GetConfigParams = GetJestConfigParams
+
+export function getConfig(opts: GetConfigParams) {
+  return configMap[opts.type](opts)
+}
```

Comment:
I'm thinking out loud here and evaluating why we might want this code.

We're giving the user discoverable single point of entry to get the configs that we've built for them in cases where they might want to extend those configs, and in the case of jest we now have to supply a configuration file with some defaults based on the build target.

`getConfig('jest', 'node')` -> `@redwoodjs/core/config/node/jest.config.js`

That seems super reasonable, and it's nice that it'll have type definitions.

I'm wondering if we could have a convention, for storing the configs, that would preclude us from having to manually map each one in this function?

```ts
export const getConfig = (type, target) => { 
   // handle exceptions with helpful warnings.
   require(`@redwoodjs/core/config/${target}/${type}.config.js`
}
```

In that we would could dynamically require the configuration and return it, without having to do any mapping.

Reply:
I'm a bit hesitant about the dynamic require because they caused me trouble in webpack in the past. I'd love for it to work tho so I'm gonna give it a try!

I'm trying to think if there are any config files that wouldn't fit into this `target/type` convention, but I can't think of any.

Reply:
I think we should be OK since at this point we're in the node context. I can't think of a reason why someone would want to use `getConfig()` in the browser context.